### ISSUE TITLE
How to list supported values of C# language versions

### ISSUE THREAD

**Issue by ghost** [dotnet/docs#27101]
Hi! 

In the first section it is mentioned that a list of supported language versions can be made/given:

_Use <LangVersion>?</LangVersion> to list supported values._

The question is how to do this? I assume I have to insert a question mark (?) in the LangVersion section of the .csproj file, but what to do next? From what I've seen in the build output is that it is not there. 
How should this list be obtained?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5026557b-74c1-888f-3362-0f1e5f35562a
* Version Independent ID: 215af505-a5c6-239e-581e-041be8483e52
* Content: [Compiler Error CS1617](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs1617)
* Content Source: [docs/csharp/misc/cs1617.md](https://github.com/dotnet/docs/blob/main/docs/csharp/misc/cs1617.md)
* Product: **dotnet-csharp**
* Technology: **csharp-diagnostics**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 454376](https://dev.azure.com/msft-skilling/Content/_workitems/edit/454376)

**Comment by BillWagner** [comment]
ping @chsienki 

I tried on the .NET 6 SDK, and the `<LangVersion>?</LangVersion>` isn't working for me now either. What should it be?

**Comment by Youssef1313** [comment]
@BillWagner I believe this works with `csc` only.

**Comment by chsienki** [comment]
@BillWagner as @Youssef1313 notes you really need to do this on csc itself. 

If you set it in the MSBuild file, it will technically work, but you'll see later build failures because the compiler outputs the langversions then returns without actually compiling anything. From a binlog you can see the output of the CSC task is listing the versions

![image](https://user-images.githubusercontent.com/16246502/142044451-48addad5-eb6d-4a1f-92f6-c13322c4f239.png)

**Comment by Nuss9** [comment]
Thanks for the replies! It looks like creating this list of compatible languages requires more than just using `<LangVersion>?</LangVersion>`. 
Maybe a good idea to edit the documentation to make it more explicit how to use this feature to prevent future questions?

Best regards!

**Comment by Youssef1313** [comment]
@BillWagner I agree with @Nuss9 that the documentation should be extended on how `<LangVersion>?</LangVersion>` can work.

**Comment by BillWagner** [comment]
I agree @Youssef1313 

I'm reopening this so we don't lose track of it.

**Comment by svick** [comment]
I was thinking the opposite: `<LangVersion>?</LangVersion>` is hard to use and not particularly useful, so I would just remove any mentions of it from the docs.

Except that the mention that exists on this page is from the [compiler error message](https://github.com/dotnet/roslyn/blob/f948cdc/src/Compilers/CSharp/Portable/CSharpResources.resx#L2864) (incorrectly changed to use `<LangVersion>` instead of `-langversion`) so it would probably require changing the actual error message.

**Comment by brandoncimino** [comment]
For anyone in the future trying to decipher what `/langversion:?` means:

- "MSBuild file" refers to multiple things, one of which is `.csproj` files.
- "csc" refers to `csc.exe`, an equivalent to `dotnet build` that is **specific to .NET Framework** (see [C# Compiler Options](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/#net-framework-projects)).
- "binlog" refers to a [binary log](https://github.com/dotnet/msbuild/blob/main/documentation/wiki/Binary-Log.md), which is _like_ a log except you need a [special tool](https://msbuildlog.com/) to look at it.
- You can log the output to a human-readable file using the command line argument [`-fileLogger`](https://docs.microsoft.com/en-us/visualstudio/msbuild/obtaining-build-logs-with-msbuild?view=vs-2022#save-the-build-log-to-a-file).

**For my .NET Standard library project**, the actual steps to I took to figure out the supported `<LangVersion>` values are:

1. Open one of my `.csproj` files.
2. Set the `<LangVersion>` attribute to `?`, i.e. `<LangVersion>?</LangVersion>`
3. From the **root project directory** (i.e. next to the `.sln` file), ran the command:
    ```
    dotnet build -fileLogger
    ```
4. The build then failed with the message:
    ```
    error MSB3030: Could not copy the file "obj\Debug\netstandard2.0\FowlFever.BSharp.dll" because it was not found.
    ```
5. Next to the `.sln` file, a new file called `msbuild.log` appeared.
6. Searching that file for `Supported language versions` turns up a list of valid `<LangVersion>` values:
    ![image](https://user-images.githubusercontent.com/5974238/177020948-6871ad93-122f-482b-aa00-e725b6fb7521.png)
